src/FluentValidation/Internal/PropertyRule.cs

```diff
@@ -115,10 +117,17 @@ TProperty PropertyFunc(T instance)
 			}
 		}
 
+		bool first = true;
+		TProperty propValue = default(TProperty);
+
 		var cascade = CascadeMode;
-		var accessor = new Lazy<TProperty>(() => PropertyFunc(context.InstanceToValidate), LazyThreadSafetyMode.None);
 		var totalFailures = context.Failures.Count;
-		context.InitializeForPropertyValidator(propertyPath, GetDisplayName, PropertyName);
+
+		if (_displayNameFunc == null) {
+			_displayNameFunc = GetDisplayName;

```

Comment:
This can be initialized inside RuleBase's constructor. Rule state shouldn't be modified inside `ValidateAsync`

Reply:
Resolved in ccc3ebe81060a1f1c9f83fa1a232e42a38060ab3